src/main/webapp/js/components/Calendar.js

```diff
@@ -1,77 +1,227 @@
   }
 
   render() {
-    const { isOpen, selectedDay, onDayClick, stats } = this.props;
+    const buildDomain = (data, slo) => {
+      const values = data && data.length > 0 && data.map(x => x.percent) || [0, 1];
+      const [min, max] = [Math.min(...values), Math.max(...values)];
+      return data && data.length > 1 && min !== max
+        ? min > slo * 0.9 ? [slo * 0.9, 1] : ['auto', 1]
+        : [0, 1];
+    };
+
+    const {isOpen, selectedDay, onDayClick, onCloseClick, stats, slo} = this.props;
+    const {flipped} = this.state;
+
+    const data = stats && Object.keys(stats.daily)
+      .map(key => ({date: Number(key), percent: stats.daily[Number(key)]}))
+      .sort((a, b) => a.date - b.date);
+
+    const wrapperClass = classNames({
+      'calendar-chart-wrapper': true,
+      closed: !isOpen,
+      flipped: flipped,
+    });
+
+    const flipperClass = classNames({
+      flipper: true,
+      flipped: flipped,
+    });
+
+    const flipViewButtonClass = classNames({
+      'flip-view-button': true,
+      chart: !flipped,
+      calendar: flipped,
+    });
+
     return (
-      <DayPicker
-        className={isOpen ? '' : 'hidden'}
-        selectedDays={selectedDay}
-        disabledDays={{
-          after: moment().startOf('day').toDate()
-        }}
-        onDayClick={ onDayClick }
-        renderDay={ this.renderDay(aggregateStatsByDay(stats)) }
-      />
+      <div className={wrapperClass}>
+          <div className={flipperClass}>
+              <Datetime className={'front'}
+                        value={selectedDay}
+                        input={false} timeFormat={false}
+                        renderDay={this.renderDay(stats && stats.daily, slo)}
+                        renderMonth={this.renderMonth(stats && stats.monthly, slo)}
+                        renderYear={this.renderYear(stats && stats.yearly, slo)}
+                        onChange={onDayClick}
+                        isValidDate={this.validateDate}
+              />
+              { data && data.length > 0 &&
+                (<ResponsiveContainer className="back" width="100%" height="100%">
+                    <BarChart data={data} margin={{left: -20}}>
+                        <XAxis dataKey="date" tickFormatter={this.xAxisTickFormatter}/>
+                        <YAxis domain={buildDomain(data, slo)}
+                               tickFormatter={this.yAxisTickFormatter}/>
+                        <Tooltip content={<CustomChartTooltip payload label active/>}/>
+                        <Brush dataKey='date' height={30} tickFormatter={this.xAxisTickFormatter}
+                               startIndex={Math.max(0, data.length - 31)}
+                               endIndex={Math.max(0, data.length - 1)}
+                               travellerWidth={10}/>
+                        <Bar dataKey="percent" shape={<ThresholdFillBar y x width height value threshold={slo}/>}/>
+                        <ReferenceLine y={slo} label="SLO"/>
```

Comment:
can we put the "SLO" title on the top of the box in white color?

Reply:
The idea was to add context to the red line itself.